apps/mobile/src/components/InsideRouter/components/SettingsScreen/actionAtoms.ts

```diff
@@ -0,0 +1,39 @@
+import {atom} from 'jotai'
+import {translationAtom} from '../../../../utils/localization/I18nProvider'
+import {pipe} from 'fp-ts/function'
+import {askAreYouSureActionAtom} from '../../../AreYouSureDialog'
+import LanguageSelect from './components/LanguageSelect'
+import * as TE from 'fp-ts/TaskEither'
+import {loadingOverlayDisplayedAtom} from '../../../LoadingOverlayProvider'
+import * as T from 'fp-ts/Task'
+import {delayInPipeT} from '../../../../utils/fpUtils'
+import {currentAppLanguageAtom, selectedLanguageAtom} from './atoms'
+
+export const changeLanguageActionAtom = atom(null, async (get, set) => {
+  const {t} = get(translationAtom)
+
+  await pipe(
+    set(askAreYouSureActionAtom, {
+      variant: 'info',
+      steps: [
+        {
+          type: 'StepWithChildren',
+          MainSectionComponent: LanguageSelect,
+          positiveButtonText: t('common.change'),
+          negativeButtonText: t('common.back'),
+        },
+      ],
+    }),
+    TE.match(
+      () => {},
+      () => {
+        set(loadingOverlayDisplayedAtom, true)
+        set(currentAppLanguageAtom, get(selectedLanguageAtom))
+      }
+    ),
+    T.chain(delayInPipeT(500)),
```

Comment:
Why is this delay here? If user declines this will still happen due to the match above.